### DISCUSSION TITLE
Add rust support in editor

### DISCUSSION THREAD

**Discussion by ABouttefeux** [godotengine/godot-proposals#4597] [Editor]
I would like to add a feature to add support in the editor to add script in rust directly using the [gdnative ](https://godot-rust.github.io/) library. The editor would interface with the rust language server to do the syntax coloration and add other feature an IDE needs.

It would also add compilation of the rust project using cargo of the script automatically and linking it when compiling the project.

I would also like to add an utility to install all tool necessary (cargo, rustc, llvm binary, rls) into the editor.

why ?

- give more access to the godot engine for dev already using rust or new dev wanting to use rust
- give an new GUI option for rust dev (which is lacking)
- I think it could be neat and I have time to implement it.

drawback

- it will require some rust code to be added to the code base which would increase its complexity.

Is this something the community wants ?

**Comment by Zireael07** [comment]
Have you tried searching? https://github.com/godot-rust/godot-rust already exists and GDNative is made exactly for this - writing more language extensions

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/3367.
